modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/GroupResourceManager.java

```diff
@@ -159,8 +159,9 @@ public SCIMResponse create(String scimObjectString, UserManager userManager,
             String encodedGroup;
             Map<String, String> httpHeaders = new HashMap<String, String>();
             if (createdGroup != null) {
-
-                encodedGroup = encoder.encodeSCIMObject(createdGroup);
+                Group copiedGroup = (Group) CopyUtil.deepCopy(createdGroup);
```

Comment:
Can you please explain the requirement to make a copy of the group object?

Reply:
I followed the pattern implemented for user resource: https://github.com/wso2/charon/blob/master/modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/UserResourceManager.java#L174

Reply:
Just by looking at this place, I don't see a value in copying the object. Can you please try to verify the implementation without copying the object.

A unit test to verify the flow is also highly appreciate.

Reply:
Agree, I simplified implementation. Also tested using `samples` module.